util/drakeUtil.h

```diff
@@ -84,19 +93,86 @@ mxArray* stdVectorToMatlab(const std::vector<Scalar>& vec) {
   return pm;
 }
 
-DLLEXPORT const std::vector<double> matlabToStdVector(const mxArray* in);
+DLLEXPORT void sizecheck(const mxArray* mat, int M, int N);
+
+template <size_t Rows, size_t Cols>
```

Comment:
In order to define a multidimensional array using [row][col], the dimensions need to be compile-time constants.